src/server/middleware/blob/put_manifest.go

```diff
@@ -0,0 +1,125 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package blob
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/docker/distribution/manifest/schema2"
+	"github.com/goharbor/harbor/src/api/blob"
+	"github.com/goharbor/harbor/src/common/utils/log"
+	"github.com/goharbor/harbor/src/pkg/distribution"
+	"github.com/goharbor/harbor/src/server/middleware"
+	"github.com/justinas/alice"
+)
+
+// PutManifestMiddleware middleware which create Blobs for the foreign layers and attach them to the project,
+// update the content type of the Blobs which already exist,
+// create Blob for the manifest, attach all Blobs to the manifest after PUT /v2/<name>/manifests/<reference> success.
+func PutManifestMiddleware() func(http.Handler) http.Handler {
+	skipper := middleware.NegativeSkipper(
```

Comment:
As you bind the middleware to the put manifest sub path in the router already, why we still need this?